UM/Settings/ContainerStack.py

```diff
@@ -391,12 +391,21 @@ def findContainer(self, criteria = None, container_type = None, **kwargs):
     #
     #   \param container The container to add to the stack.
     def addContainer(self, container):
-        if container is not self:
-            container.propertyChanged.connect(self._collectPropertyChanges)
-            self._containers.insert(0, container)
-            self.containersChanged.emit(container)
-        else:
+        self.insertContainer(0, container)
+
+    ##  Insert a container into the stack.
+    #
+    #   \param index \type{int} The index of to insert the container at.
+    #          A negative index counts from the bottom
+    #   \param container The container to add to the stack.
+    def insertContainer(self, index, container):
+        if container is self:
             raise Exception("Unable to add stack to itself.")
+            return
```

Comment:
An exception will still exit the running code. What it does not do, is abort the main loop. So yes, the return is superfluous.